aten/src/ATen/native/cuda/fused_adam_utils.cuh

```diff
@@ -17,20 +18,25 @@ constexpr uint8_t kExpAvgIdx = 2;
 constexpr uint8_t kExpAvgSqIdx = 3;
 constexpr uint8_t kMaxExpAvgSqIdx = 4;
 
-template <typename scalar_type, typename opmath_t, int depth = 4>
-C10_DEVICE __forceinline__ void adam_math(
+template <
```

Comment:
Again for my understanding: Does moving stuff to templates increase binary size to allow for better code inlining? How come this matters for performance?